server/src/controllers/coursesection.ts

```diff
@@ -1,4 +1,45 @@
 /**
  * TODO: Add controller functions which will be triggered when /api/coursesection endpoint is hit
- * SEE prereq controller for examples
+ * SEE sections controller for examples
  */
+import parentLogger from "../../utils/logger";
+import db from "../database/db";
+import {Request, Response} from "express";
+
+const log = parentLogger.child({ module: "router/coursesection" });
+
+/**
+ * The following functions are triggered when requests to /api/sections* are made
+ */
+
+export const getSections = async (req: Request, res: Response) => {
+    log.info("GET api/sections");
+    try {
+        const { rows } = await db.query("SELECT * FROM coursesection");
+        res.status(200).json(rows);
+    } catch (e) {
+        return res.status(400).json({
+            message: e.message
+        });
+    }
+};
+
+export const getSection = async (req: Request, res: Response) => {
```

Comment:
Is an error thrown when either `coursedept` and/or `coursenumber` are not specified? If not, I think we should handle this case and perhaps make these path parameters required?

Reply:
We dont need that since without either parameter, you get a Cannot GET route error.